test/unittests/commands/create/create_on_server/test_properties.py

```diff
@@ -0,0 +1,364 @@
+from dsp_tools.commands.create.models.create_problems import UploadProblemType
+from dsp_tools.commands.create.models.parsed_ontology import GuiElement
+from dsp_tools.commands.create.models.parsed_ontology import KnoraObjectType
+from dsp_tools.commands.create.models.parsed_ontology import ParsedProperty
+from dsp_tools.commands.create.models.server_project_info import CreatedIriCollection
+from dsp_tools.commands.create.models.server_project_info import ListNameToIriLookup
+from dsp_tools.commands.create.models.server_project_info import OntoCreateLookup
+from dsp_tools.commands.create.models.server_project_info import ProjectIriLookup
+from dsp_tools.utils.request_utils import ResponseCodeAndText
+from test.unittests.commands.create.constants import LAST_MOD_2
+from test.unittests.commands.create.constants import LAST_MOD_3
+from test.unittests.commands.create.constants import LAST_MODIFICATION_DATE
+from test.unittests.commands.create.constants import ONTO
+from test.unittests.commands.create.constants import ONTO_IRI
+from test.unittests.commands.create.constants import PROJECT_IRI
+
+KNORA_SUPER = "http://api.knora.org/ontology/knora-api/v2#hasValue"
+EXTERNAL_SUPER = "http://xmlns.com/foaf/0.1/name"
+FAILED_CLASS = "http://api.knora.org/ontology/0001/onto/v2#FailedClass"
+
+
+def make_test_property(name: str, supers: list[str] | None = None) -> ParsedProperty:
+    """Helper function to create a test property with standard defaults."""
+    supr = supers if supers else []
+    supr.append(KNORA_SUPER)
+    return ParsedProperty(
+        name=str(ONTO[name]),
+        labels={"en": f"Label for {name}"},
+        comments=None,
+        supers=supr,
+        object=KnoraObjectType.TEXT,
+        subject=None,
+        gui_element=GuiElement.SIMPLETEXT,
+        node_name=None,
+        onto_iri=str(ONTO_IRI),
+    )
+
+
+@pytest.fixture
+def three_multiple_inheritance_props():
+    """PropA inherits from both PropB and PropC (both independent)."""
+    prop_b = make_test_property("PropB")
+    prop_c = make_test_property("PropC")
+    prop_a = make_test_property("PropA", supers=[prop_b.name, prop_c.name])
+    return [prop_a, prop_b, prop_c]
+
+
+@pytest.fixture
+def three_independent_props():
+    """PropA inherits from both PropB and PropC (both independent)."""
+    prop_b = make_test_property("PropB")
+    prop_c = make_test_property("PropC")
+    prop_a = make_test_property("PropA")
+    return [prop_a, prop_b, prop_c]
+
+
+class TestGetPropertyOrder:
+    def test_multiple_inheritance_scenario(self, three_multiple_inheritance_props):
+        prop_a, _, _ = three_multiple_inheritance_props
+        result = _get_property_create_order(three_multiple_inheritance_props)
+        # A has edges to both B and C, so A comes before both
+        # the order afterwards does not matter
+        assert len(result) == 3
+        assert result[-1] == prop_a.name
+
+    def test_external_supers_do_not_break_sorting(self):
```

Comment:
You don't have an external super here. KNORA_SUPER is always added, so this is already tested in all other tests.